### PR TITLE
Add support for Ruby 2.5 - 3.0

### PR DESCRIPTION
**Pull request by HertzDevil**
[Ruby](https://www.ruby-lang.org/en/) is...

> A dynamic, open source programming language with a focus on simplicity and productivity. It has an elegant syntax that is natural to read and easy to write.

This PR is largely inspired from the current Python support. The editor syntax is borrowed from Monaco and a minimal one is made for the disassembly output.

Related: compiler-explorer/infra#563

### CODE REVIEW

File: etc/config/ruby.amazon.properties
Comment (supergrecko) [inline etc/config/ruby.amazon.properties:17]: Looks good! I think it'd be conventional to name the compilers ruby274 for 2.7.4. There could be interest in installing versions like 2.7.3 in the future and this would help to avoid name collision.

```suggestion
defaultCompiler=ruby302

group.ruby.compilers=ruby302:ruby274:ruby268:ruby259
group.ruby.isSemVer=true
group.ruby.baseName=Ruby
group.ruby.groupName=Ruby YARV
group.ruby.compilerType=ruby

compiler.ruby302.semver=3.0.2
compiler.ruby302.exe=/opt/compiler-explorer/ruby-3.0.2/bin/ruby
compiler.ruby274.semver=2.7.4
compiler.ruby274.exe=/opt/compiler-explorer/ruby-2.7.4/bin/ruby
compiler.ruby268.semver=2.6.8
compiler.ruby268.exe=/opt/compiler-explorer/ruby-2.6.8/bin/ruby
compiler.ruby259.semver=2.5.9
compiler.ruby259.exe=/opt/compiler-explorer/ruby-2.5.9/bin/ruby
```

File: etc/scripts/ruby/disasm.rb
Comment (RubenRBS) [inline etc/scripts/ruby/disasm.rb:1]: I wonder if https://github.com/compiler-explorer/tools is a better place for this. I'll ask

**Review by RubenRBS — COMMENTED** [review]
Can you add a line to https://github.com/compiler-explorer/compiler-explorer/blob/main/.github/labeler.yml too? :)

File: etc/scripts/ruby/disasm.rb
Comment (RubenRBS) [reply etc/scripts/ruby/disasm.rb:1]: I wonder no more. We already use a python script for a similar pourpose, so I guess this is ok! :)

**Review by apmorton — APPROVED** [review]

**Review by apmorton — CHANGES_REQUESTED** [review]
no changes requested here, just marking disapproval so other maintainers don't merge just yet.

There are some errors installing ruby 2.7.4, and I want to sort out the builder stuff before merging this.

**Comment by partouf** [comment]
this is now live

**Pull request merged**
